extension/src/helpers/muxedAddress.ts

```diff
@@ -0,0 +1,228 @@
+/**
+ * Utility functions for handling muxed addresses and memo logic
+ */
+import { NetworkDetails } from "@shared/constants/stellar";
+import { getContractSpec } from "@shared/api/internal";
+import { isContractId } from "popup/helpers/soroban";
+import {
+  isMuxedAccount,
+  isValidStellarAddress,
+  createMuxedAccount,
+  getBaseAccount,
+  isFederationAddress,
+} from "helpers/stellar";
+import { StrKey } from "stellar-sdk";
+
+export interface CheckContractMuxedSupportParams {
+  contractId: string;
+  networkDetails: NetworkDetails;
+}
+
+/**
```

Comment:
let's add tests for the helpers in this file. We're covering a lot of different scenarios so it'd be good to have some coverage here